core/src/main/java/pt/webdetails/cgg/SVGChart.java

```diff
@@ -23,59 +26,123 @@
 import org.apache.batik.transcoder.image.PNGTranscoder;
 import org.apache.batik.transcoder.svg2svg.SVGTranscoder;
 import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
 import pt.webdetails.cpf.utils.CharsetHelper;
 
+import javax.imageio.ImageIO;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+
 /**
  * @author pdpi
  */
-public class SVGChart implements Chart
-{
+public class SVGChart implements Chart {
   private Document svg;
+  private boolean isMultiPage;
 
-  public SVGChart(final Document doc)
-  {
-    if (doc == null)
-    {
+  public SVGChart( final Document doc, final boolean isMultiPage ) {
+    if ( doc == null ) {
       throw new NullPointerException();
     }
     this.svg = doc;
+    this.isMultiPage = isMultiPage;
   }
 
-  public Document getRawObject()
-  {
+  public Document getRawObject() {
     return svg;
   }
 
-  public void renderAsPng(final OutputStream out) throws IOException
-  {
+  public void renderAsPng( final OutputStream out ) throws IOException {
+    if ( this.isMultiPage ) {
+      renderPages( out, false );
+    } else {
+      renderPageAsPng( out, svg );
+    }
+  }
+
+  public void renderAsSVG( final OutputStream out ) throws IOException {
+    if ( this.isMultiPage ) {
+      renderPages( out, true );
+    } else {
+      renderPageAsSVG( out, svg );
+    }
+  }
+
+  private void renderPages( final OutputStream out, boolean isAsSvg ) throws IOException {
+    // Each child element of svg's document element represents a page.
+    // Remove all but one of those children and render each page.
+    Element rootElem = svg.getDocumentElement();
+    NodeList pageNodeList = rootElem.getChildNodes();
+    int pageCount = pageNodeList.getLength();
+
+    // Backup page nodes in a node array.
+    Node[] pageNodes = new Node[pageCount];
+    for ( int pageIndex = 0; pageIndex < pageCount; pageIndex++ ) {
+      pageNodes[pageIndex] = pageNodeList.item( pageIndex );
+    }
+
+    // Remove all page nodes.
+    while ( rootElem.getLastChild() != null ) {
+      rootElem.removeChild( rootElem.getLastChild() );
+    }
+
+    // Create a "mixed" multipart.
+    MimeMultipart multiPart = new MimeMultipart();
+
+    try {
+      // Now add one child node in turn.
+      for ( int pageIndex = 0; pageIndex < pageCount; pageIndex++ ) {
+        rootElem.appendChild( pageNodes[pageIndex] );
+
+        ByteArrayOutputStream pageOutputStream = new ByteArrayOutputStream();
+        if ( isAsSvg ) {
+          renderPageAsSVG( pageOutputStream, svg );
+        } else {
+          renderPageAsPng( pageOutputStream, svg );
+        }
+
+        rootElem.removeChild( pageNodes[pageIndex] );
+
+        BufferedImage image = ImageIO.read( new ByteArrayInputStream( pageOutputStream.toByteArray() ) );
```

Comment:
Not sure why we need to parse the image. In the non-multipage scenario we just pass the binary stream created by respective Transcoder... Why can't it be simply placed in the body part?